lib/reporter/tap.js

```diff
@@ -12,7 +12,11 @@ function generateTest(mod, count) {
     mod.error = mod.error.message;
   }
   var error = mod.error ? [directive, mod.error].join(' ') : '';
-  var output = mod.testOutput ? '\n' + util.sanitizeOutput(mod.testOutput, '# ') : '';
+  if (mod.testOutput && mod.testOutput.length > 0) {
+    var output = '\n  ---\n' + util.sanitizeOutput(mod.testOutput, '  #') + '\n  ...';
```

Comment:
I think the question is whether TAP parsers will ignore `ok` lines within YAMLish blocks. I'll have a look with @GeorgeAdams95 to see what Jenkins makes of it.